index.html

```diff
@@ -1806,30 +1809,49 @@
                   </ol>
                 </li>
                 <li>If <var>previousAvailability</var> is not equal to
-                <var>newAvailability</var>, then <a>queue a task</a> to run the
-                following steps:
+                <var>newAvailability</var>, then run the following steps:
                   <ol>
-                    <li>Set <var>A</var>'s <a>value</a> property to
-                    <var>newAvailability</var>.
+                    <li>If <var>A</var> has already been exposed to ECMAScript
```

Comment:
> I would like to not rely on "exposed to ECMAscript" to define behavior since (as an implementer) I'm not entirely sure what that means.

OK, let's try to get rid of it.

> The first time the monitoring algorithm runs for A, the promise for A should be unsettled, correct? Could you fire the event only after the availability promise is resolved?

That would work but I do not know how to introduce the availability promise at this step of the monitoring algorithm: "If there is an unsettled Promise from the call to `getAvailability` that created A" seems obscure. Or not?

> Could step 8 of getAvailability initialize value to undefined, and then the event would fire only when the previous value was set?

Problem is the `value` property is a `boolean` which cannot be `undefined`. Or perhaps we can still say "If A's value has not been initialized yet" (and drop steps that initialize "value" elsewhere) and let this as an implementation exercise to track whether initialization occurred already?